chaoscenter/graphql/server/pkg/database/mongodb/environments/operations.go

```diff
@@ -25,6 +25,18 @@ func NewEnvironmentOperator(mongodbOperator mongodb.MongoOperator) *Operator {
 	}
 }
 
+// Inside the dbOperationsEnvironment package
+
+type EnvironmentOperatorInterface interface {
+	InsertEnvironment(ctx context.Context, env Environment) error
+	GetEnvironments(ctx context.Context, query bson.D) ([]Environment, error)
+	UpdateEnvironment(ctx context.Context, query bson.D, update bson.D) error
+	GetEnvironment(query bson.D) (Environment, error)
+	GetAggregateEnvironments(pipeline mongo.Pipeline) (*mongo.Cursor, error)
+	GetEnvironmentDetails(ctx context.Context, environmentID string, projectID string) (Environment, error)
+	GetEnvironmentWithProjectID(projectID string) ([]*Environment, error)
+}
+
 // InsertEnvironment takes details of a chaos_environment and inserts into the database collection
```

Comment:
IMO, we don't need this code if we adopt below code
```golang
import (
  dbMocks "github.com/litmuschaos/litmus/chaoscenter/graphql/server/pkg/database/mongodb/mocks"
  dbOperationsEnvironment "github.com/litmuschaos/litmus/chaoscenter/graphql/server/pkg/database/mongodb/environments"
)

mongodbMockOperator := new(dbMocks.MongoOperator)
environmentOperator := dbOperationsEnvironment.NewEnvironmentOperator(mongodbMockOperator)
```

we don't also need `MockEnvironmentOperator` struct and relevant functions in `chaoscenter/graphql/server/pkg/environment/handler/handler_test.go` if we adopt the above.

I think reducing unnecessary code is a great way to make open source understandable to many people.

How do you think @Saranya-jena @gdsoumya @Freedisch ?

Reply:
Make sense. I updated the code to reflect it